### PR TITLE
Updated the Habitat 2.0 citation

### PR DESCRIPTION
**Pull request by ASzot**
## Motivation and Context

The recently released Habitat 2.0 paper needs to be in the references and citations list. 

## How Has This Been Tested

Visually inspected rendered README. 

## Types of changes

- Docs change / refactoring / dependency upgrade

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by mathfac — APPROVED** [review]
Great, let's add bibtex as well.

**Comment by dhruvbatra** [comment]
@ASzot -- same comment. Please also add bibtex to the "Citing Habitat" section.

**Comment by ASzot** [comment]
Now added.

File: README.md [hunk 1]
```diff
@@ -47,7 +47,7 @@ While there has been significant progress in the vision and language communities
 To this end, we aim to standardize the entire ‘software stack’ for training embodied agents – scanning the world and creating highly photorealistic 3D assets, developing the next generation of highly efficient and parallelizable simulators, specifying embodied AI tasks that enable us to benchmark scientific progress, and releasing modular high-level libraries to train and deploy embodied agents.
 
 ## Citing Habitat
-If you use the Habitat platform in your research, please cite the following [paper](https://arxiv.org/abs/1904.01201):
+If you use the Habitat platform in your research, please cite the [Habitat](https://arxiv.org/abs/1904.01201) and [Habitat 2.0](https://arxiv.org/abs/2106.14405) papers:
```
Comment (Skylion007) [inline README.md:50]: Shouldn't you reorder them? Always best to encourage them to cite the newest paper.

Comment (Skylion007) [reply README.md:50]: Can we make ti even easier and put the bibtex for it here?

File: README.md [hunk 2]
```diff
@@ -56,6 +56,12 @@ If you use the Habitat platform in your research, please cite the following [pap
   booktitle =     {Proceedings of the IEEE/CVF International Conference on Computer Vision (ICCV)},
   year      =     {2019}
 }
+@article{szot2021habitat,
```
Comment (Skylion007) [inline README.md:59]: I personally like the newest paper first style. Thoughts @dhruvbatra @mathfac ?

Comment (mathfac) [reply README.md:59]: That makes sense to me.

Comment (dhruvbatra) [reply README.md:59]: Sounds good to me too.

**Pull request merged**